api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -145,4 +151,50 @@
    */
   @Deprecated
   OverwriteFiles validateNoConflictingAppends(Long readSnapshotId, Expression conflictDetectionFilter);
+
+  /**
+   * Sets a conflict detection filter used to validate concurrently added data and delete files.
+   * <p>
+   * If not called, a true literal will be used as the conflict detection filter.
```

Comment:
This isn't quite accurate because the row filter will be used.

Reply:
It described how it worked before. I've changed the logic and removed this sentence too.